Files/Filesystem/Search/FolderSearch.cs

```diff
@@ -204,7 +204,7 @@ private async Task SearchTagsAsync(string folder, IList<ListedItem> results, Can
                     return (hFileTsk, findDataTsk);
                 }).WithTimeoutAsync(TimeSpan.FromSeconds(5));
 
-                if (hFile != IntPtr.Zero)
+                if (hFile != IntPtr.Zero && hFile.ToInt64() != -1)
```

Comment:
Is `hFile != IntPtr.Zero` needed?

Reply:
Yeah because of the timeout above, if the timeout expires it returns the default value for an IntPtr which is IntPtr.Zero